interfaces/inc/file.h

```diff
@@ -0,0 +1,39 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+#ifndef SYNC_H
+#define SYNC_H
+
+#ifdef __cplusplus
+#include <cstdbool>
+#else
+#include <stdbool.h>
+#endif
+
+#include "umock_c/umock_c_prod.h"
+#include "constbuffer.h"
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+#define FILE_DATA_RESULT_VALUES \
+    FILE_DATA_FAILURE, \
+    FILE_DATA_INVALID, \
+    FILE_DATA_OK
+
+MU_DEFINE_ENUM(FILE_DATA_RESULT, FILE_DATA_RESULT_VALUES);
+
+typedef void *FILE_HANDLE;
+typedef void(*FILE_WRITE_CB)(void* userContext, bool isSuccess);

```

Comment:
>bool [](start = 48, length = 4)

Consider having an enum so it is symmetric with read

Reply:
I changed both to bool instead. Is that okay?


---
In reply to: [445223280](https://github.com/Azure/azure-c-pal/pull/7#discussion_r445223280) [](ancestors = 445223280)